packages/svelte/src/internal/client/dom/blocks/each.js

```diff
@@ -257,7 +259,10 @@ export function each(node, flags, get_collection, get_key, render_fn, fallback_f
 		// that a mutation occurred and it's made the collection MAYBE_DIRTY, so reading the
 		// collection again can provide consistency to the reactive graph again as the deriveds
 		// will now be `CLEAN`.
-		get(each_array);
+		if (first_run) {
```

Comment:
I wonder if it's only first run. Like if you run a conditional collection that changes deps between runs, is that also a problem?

Reply:
It's a shame we use `derived_safe_equal` otherwise this wouldn't be an issue.